rllib/algorithms/ppo/torch/ppo_torch_rl_module.py

```diff
@@ -70,147 +77,158 @@ def get_separate_encoder_config(env):
+                num_layers=1,
+            )
+        else:
+            encoder_config = FCConfig(
+                hidden_layers=fcnet_hiddens,
+                activation=activation,
+                output_dim=model_config["fcnet_hiddens"][-1],
+            )
 
-class PPOTorchRLModule(TorchRLModule):
-    def __init__(self, config: PPOModuleConfig) -> None:
-        super().__init__(config)
+        pi_config = FCConfig()
+        vf_config = FCConfig()
 
-    def setup(self) -> None:
         assert isinstance(
-            self.config.observation_space, gym.spaces.Box
+            observation_space, gym.spaces.Box
         ), "This simple PPOModule only supports Box observation space."
 
         assert (
-            len(self.config.observation_space.shape) == 1
+            len(observation_space.shape) == 1
         ), "This simple PPOModule only supports 1D observation space."
 
-        assert isinstance(
-            self.config.action_space, (gym.spaces.Discrete, gym.spaces.Box)
-        ), ("This simple PPOModule only supports Discrete and Box action space.",)
-
-        assert self.config.pi_config, "pi_config must be provided."
-        assert self.config.vf_config, "vf_config must be provided."
-
-        self.encoder = None
-        if self.config.encoder_config:
-            encoder_config = self.config.encoder_config
-            encoder_config.input_dim = self.config.observation_space.shape[0]
-            self.encoder = FCNet(self.config.encoder_config)
+        assert isinstance(action_space, (gym.spaces.Discrete, gym.spaces.Box)), (
+            "This simple PPOModule only supports Discrete and Box action space.",
+        )
 
         # build pi network
-        pi_config = self.config.pi_config
-        if not self.encoder:
-            pi_config.input_dim = self.config.observation_space.shape[0]
-        else:
-            pi_config.input_dim = self.encoder.output_dim
+        encoder_config.input_dim = observation_space.shape[0]
+        pi_config.input_dim = encoder_config.output_dim
 
-        if isinstance(self.config.action_space, gym.spaces.Discrete):
-            pi_config.output_dim = self.config.action_space.n
+        if isinstance(action_space, gym.spaces.Discrete):
+            pi_config.output_dim = action_space.n
         else:
-            pi_config.output_dim = self.config.action_space.shape[0] * 2
-        self.pi = FCNet(pi_config)
+            pi_config.output_dim = action_space.shape[0] * 2
 
         # build vf network
-        vf_config = self.config.vf_config
-        if not self.encoder:
-            vf_config.input_dim = self.config.observation_space.shape[0]
-        else:
-            vf_config.input_dim = self.encoder.output_dim
-
+        vf_config.input_dim = encoder_config.output_dim
         vf_config.output_dim = 1
-        self.vf = FCNet(vf_config)
 
-        self._is_discrete = isinstance(self.config.action_space, gym.spaces.Discrete)
+        config_ = PPOModuleConfig(
```

Comment:
why postfix _?

Reply:
it's implementation details? no need for external user to rely on this attribute.